vm/systemSmartContracts/esdt.go

```diff
@@ -251,26 +254,51 @@ func (e *esdt) issueToken(owner []byte, arguments [][]byte) error {
 	return nil
 }
 
-func (e *esdt) upgradeProperties(token *ESDTData, args [][]byte) {
-	for _, arg := range args {
-		optionalArg := string(arg)
+func upgradeProperties(token *ESDTData, args [][]byte) error {
+	if len(args) == 0 {
+		return nil
+	}
+	if len(args)%2 != 0 {
+		return vm.ErrInvalidNumOfArguments
+	}
+
+	for i := 0; i < len(args); i += 2 {
+		optionalArg := string(args[i])
+		val, err := checkAndGetSetting(string(args[i+1]))
+		if err != nil {
+			return err
+		}
 		switch optionalArg {
 		case burnable:
-			token.Burnable = true
+			token.Burnable = val
 		case mintable:
-			token.Mintable = true
+			token.Mintable = val
 		case canPause:
-			token.CanPause = true
+			token.CanPause = val
 		case canFreeze:
-			token.CanFreeze = true
+			token.CanFreeze = val
 		case canWipe:
-			token.CanWipe = true
+			token.CanWipe = val
 		case upgradable:
-			token.Upgradable = true
+			token.Upgradable = val
 		case canChangeOwner:
-			token.CanChangeOwner = true
+			token.CanChangeOwner = val
+		default:
+			return vm.ErrInvalidArgument
 		}
 	}
+
+	return nil
+}
+
+func checkAndGetSetting(arg string) (bool, error) {
+	if arg == "true" {
```

Comment:
Might use something like:
```GO
if strings.ToLower(arg) == "true" {.....

if strings.ToLower(arg) == "false" {.
```

as to work regardless how the caps were used

Reply:
wouldn't change it - it has to be strict.